lib/blocks.php

```diff
@@ -148,119 +148,55 @@ function get_dynamic_blocks_regex() {
  * Renders a single block into a HTML string.
  *
  * @since 1.9.0
+ * @since 4.4.0 renders full nested tree of blocks before reassembling into HTML string
+ * @global WP_Post $post The post to edit.
  *
  * @param  array $block A single parsed block object.
  * @return string String of rendered HTML.
  */
 function gutenberg_render_block( $block ) {
-	$block_name  = isset( $block['blockName'] ) ? $block['blockName'] : null;
-	$attributes  = is_array( $block['attrs'] ) ? $block['attrs'] : array();
-	$raw_content = isset( $block['innerHTML'] ) ? $block['innerHTML'] : null;
+	global $post;
 
-	if ( $block_name ) {
-		$block_type = WP_Block_Type_Registry::get_instance()->get_registered( $block_name );
-		if ( null !== $block_type && $block_type->is_dynamic() ) {
-			return $block_type->render( $attributes );
-		}
+	$block_type    = WP_Block_Type_Registry::get_instance()->get_registered( $block['blockName'] );
+	$is_dynamic    = $block['blockName'] && null !== $block_type && $block_type->is_dynamic();
+	$inner_content = '';
+	$index         = 0;
+
+	foreach ( $block['innerContent'] as $chunk ) {
```

Comment:
Also a number of `Undefined index:  innerContent` errors on the 7.1 test suite.

Try guarding with:
```php
isset( $block['innerContent'] ) 
```

Reply:
this is strange. if `innerContent` is unavailable then everything else should be broken. I'm stunned; will look into this